### PR TITLE
feat(analytics): ConsentService localStorage + onboarding consent + settings opt-out (Batch 3b)

### PR DESCRIPTION
**Pull request by pannpers**
## Summary

Second frontend batch for the [introduce-analytics-tool](https://github.com/liverty-music/specification/tree/main/openspec/changes/introduce-analytics-tool) change. Replaces the [Batch 3a](https://github.com/liverty-music/frontend/pull/379) fail-closed ConsentService stub with a real localStorage-backed implementation, adds a "trust commitment" consent step as the final phase of the onboarding flow, exposes the same two toggles in settings, and wires AnalyticsService to react to consent transitions by swapping PostHog's persistence + opt-in posture.

| Capability | Where |
| --- | --- |
| `ConsentService` (localStorage + `ConsentChanged` event) | `src/lib/consent/consent-service.ts` + `consent-changed.ts` |
| `AnalyticsService` consent reaction | `src/lib/analytics/analytics-service.ts` |
| Consent route (final onboarding step) | `src/routes/consent/consent-route.{ts,html,css}` |
| Settings "Privacy & analytics" section | `src/routes/settings/settings-route.{ts,html}` |
| i18n (en + ja) | `src/locales/{en,ja}/translation.json` |
| Onboarding step `CONSENT` | `src/entities/onboarding.ts` + `services/onboarding-service.ts` |

## Design highlights

- **Versioned localStorage**: `liverty:consent:state:v1` with a separate `liverty:consent:deferred:v1` flag for the "set up later" path. Corrupt JSON or schema mismatch on read is logged + recovered to defaults; the bad blob is removed so the next write isn't blocked.
- **Synchronous SDK posture**: Every mutator (`grant`, `revoke`, `defer`) persists AND publishes `ConsentChanged` via `IEventAggregator`. `AnalyticsService` subscribes in its constructor; the SDK config swap happens within the same task that the user toggled the switch.
- **Grant path**: `posthog.opt_in_capturing()` + `set_config({ persistence: 'localStorage+cookie', ip: true })`. Subsequent captures persist across reloads with a stable distinct_id.
- **Revoke path**: `posthog.opt_out_capturing()` + `set_config({ persistence: 'memory', ip: false })` + `posthog.reset()`. The reset is the important one — it severs any locally stored distinct_id, so a user who granted then revoked is back to the anonymous-memory-only posture immediately.
- **Pre-init buffering**: A `pendingConsent` field absorbs `ConsentChanged` events that fire before `posthog.init` resolves. Applied inside the init callback after the SDK loads, so an early grant in the same boot as construction takes effect for the queue replay.
- **Onboarding integration**: `STEP_ORDER` becomes `[lp, discovery, dashboard, my-artists, consent, completed]`. Existing stored value of `'completed'` stays `'completed'` — no regression for users who finished onboarding before this change.
- **Three actions on the consent screen**: "Accept selected" (per-toggle grant), "Decline all" (revoke both), "Set up later" (defer). All three call `onboarding.complete()` then route to `/dashboard`, so the step is never sticky.

## Privacy posture framing

All consent-screen and settings copy frames PostHog Cloud EU + APPI Article 28 adequacy designation as a **trust commitment beyond the statutory baseline** — NOT as legal compliance. Per project [research](https://github.com/liverty-music/specification/blob/main/docs/analytics/consent-architecture-survey.html), GDPR Article 7 and APPI explicit-consent rules do not apply to a Japan-resident-focused small-team product. The consent UX is a deliberate design choice, not a regulatory mandate.

## Scope notes

- Out of scope (Batch 3c): per-event instrumentation (artist-discovery, concert-detail, ticket-lottery, ticket-purchase, push), `UserHydrationTask` identify-on-login wiring.
- Out of scope (cloud-provisioning): the `posthogProjectKey` ConfigMap update — until that lands, AnalyticsService stays in nil-config debug-log mode.

## Tests

- `consent-service.spec.ts` — 9 new cases: hydration (defaults / valid / corrupt / schema-mismatch / deferred-flag), grant + idempotence, revoke, defer + idempotence, `ConsentChanged` publish.
- `analytics-service.spec.ts` — 2 new cases: grant → `opt_in_capturing` + persistence upgrade (no `reset()`); revoke → `opt_out_capturing` + `set_config({ persistence: 'memory' })` + `reset()`.
- `onboarding.spec.ts` — updated `stepIndex` (consent=4, completed=5) and `isOnboarding` expectations.

## Test plan

- [x] `make check` passes locally (biome lint + stylelint + typecheck + 1111 vitest tests + 7 script tests + brand-vocabulary)
- [ ] CI green
- [ ] Manual verification: walk through onboarding to the consent step, toggle both purposes, accept, observe `liverty:consent:state:v1` populated, posthog `__ph_persistence` set to `'localStorage+cookie'`; revisit settings, toggle off, observe persistence drops back to `'memory'` and `posthog.get_distinct_id()` regenerates.

Refs: liverty-music/specification#532

### CODE REVIEW

File: src/lib/consent/consent-service.ts [hunk 1]
```diff
@@ -50,66 +51,226 @@ export interface IConsentService {
...
+
+	// -- Internals --------------------------------------------------------
+
+	private applyMutation(purpose: ConsentPurpose, value: boolean): void {
+		// Short-circuit no-op mutations: granting an already-granted
+		// purpose MUST NOT republish ConsentChanged (AnalyticsService
+		// reads `set_config` as a side effect, so spurious republishes
+		// would drive redundant SDK reconfiguration on every settings
+		// page mount).
+		if (this.state[purpose] === value) return
```
[the remark this answers is not in the document]
Comment (pannpers) [reply src/lib/consent/consent-service.ts]: ✅ Fixed in 9e518c8 along the lines you suggested.

`applyMutation` now distinguishes **three branches**:

```typescript
private applyMutation(purpose: ConsentPurpose, value: boolean): void {
    const isNoOp = this.state[purpose] === value
    if (isNoOp && this.decidedAt !== null) return

    if (!isNoOp) {
        this.state = { ...this.state, [purpose]: value }
    }
    this.decidedAt = new Date().toISOString()
    if (!saveConsentState(this.state, this.decidedAt)) {
        this.logger.warn('Failed to persist consent state to localStorage')
    }
    if (!isNoOp) {
        this.ea.publish(new ConsentChanged(this.state))
    }
    this.logger.info('Consent state changed', { purpose, value })
}
```

- **Value changed**: persist + publish + log (unchanged from before).
- **Value unchanged AND not yet decided**: persist + log, **SKIP publish**. Fresh users hitting "Decline all" now write `decidedAt` to localStorage so `hasDecided()` flips to true on the next boot. AnalyticsService receives no `ConsentChanged` because its SDK posture (memory persistence + opt-out) is identical before and after — re-firing `set_config` would be wasteful.
- **Value unchanged AND already decided**: full no-op. Returning users toggling settings back to a value they already hold do not refresh `decidedAt` (no spurious timestamp update) and do not republish.

Two regression tests pin both edge cases in [`consent-service.spec.ts`](https://github.com/liverty-music/frontend/blob/9e518c8/src/lib/consent/consent-service.spec.ts):

1. `stamps decidedAt on first explicit decline even when value already matches default` — verifies the bug you found is fixed.
2. `is a full no-op when value is unchanged AND user already decided` — verifies the returning-user idempotency you correctly identified as legitimate is preserved.

Same `applyMutation` covers `acceptSelected()` with both toggles already at their value, and Settings-page toggles by never-decided users — all three call sites benefit from the first-decision branch.

File: src/lib/consent/consent-service.ts [hunk 2]
```diff
@@ -50,66 +51,226 @@ export interface IConsentService {
...
+
+	private persist(): void {
+		const payload: StoredStateV1 = {
+			version: 1,
+			analytics: this.state.analytics,
+			marketingMeasurement: this.state.marketingMeasurement,
+			decidedAt: this.decidedAt,
+		}
+		try {
+			localStorage.setItem(LS_KEY_STATE, JSON.stringify(payload))
```
[the remark this answers is not in the document]
Comment (pannpers) [reply src/lib/consent/consent-service.ts]: ✅ Fixed in 9e518c8.

Created [`src/adapter/storage/consent-storage.ts`](https://github.com/liverty-music/frontend/blob/9e518c8/src/adapter/storage/consent-storage.ts) following the same pattern as `onboarding-storage.ts` and `guest-storage.ts`:

```typescript
export function loadConsentState(): StoredConsentStateV1 | null
export function saveConsentState(state: ConsentState, decidedAt: string | null): boolean
export function loadConsentDeferred(): boolean
export function saveConsentDeferred(): boolean
export function clearConsentStorage(): void  // for tests + future "reset all"
```

All consent-related `localStorage` calls now live exclusively in this adapter:
- The two key constants (`liverty:consent:state:v1`, `liverty:consent:deferred:v1`).
- The `StoredConsentStateV1` schema type.
- `safeParse` JSON-schema validation.
- Corrupt-blob recovery (`localStorage.removeItem` on the bad blob so the warning surface does not repeat on every boot).
- `SecurityError` tolerance (privacy-mode Safari, sandboxed iframes) — load functions return `null`/`false`, save functions return `false`, all swallowed by the caller.

[`ConsentService`](https://github.com/liverty-music/frontend/blob/9e518c8/src/lib/consent/consent-service.ts) is now ~113 lines lighter — no direct `localStorage` calls, no key constants, no schema validation. The service is portable to a different backing store (IndexedDB, server-mirrored profile) without touching service code, matching the abstraction CLAUDE.md Key Decision #3 specifies.

All 12 ConsentService tests still green, plus the 2 new regression tests from the sibling fix [#3328539570](https://github.com/liverty-music/frontend/pull/380#discussion_r3328539570).

File: src/entities/onboarding.ts [hunk 3]
```diff
@@ -21,6 +29,7 @@ export const STEP_ORDER = [
 	OnboardingStep.DISCOVERY,
 	OnboardingStep.DASHBOARD,
 	OnboardingStep.MY_ARTISTS,
+	OnboardingStep.CONSENT,
 	OnboardingStep.COMPLETED,
```
[the remark this answers is not in the document]
Comment (pannpers) [reply src/entities/onboarding.ts:33]: ✅ Fixed in 9bdd79d. Good catch — the consent route was indeed dead code on the normal happy path.

Changed [`my-artists-route.ts:259`](https://github.com/liverty-music/frontend/blob/9bdd79d/src/routes/my-artists/my-artists-route.ts#L256-L266) to advance to `OnboardingStep.CONSENT` instead of `COMPLETED`:

```typescript
if (this.isOnboardingStepMyArtists) {
    this.prevHypes.set(artistId, hype)
    this.onboarding.deactivateSpotlight()
    this.onboarding.setStep(OnboardingStep.CONSENT)
    // ...
}
```

The consent route itself calls `onboarding.complete()` in all three exit paths (`acceptSelected`, `declineAll`, `setUpLater`), so the final transition to COMPLETED still happens — just one screen later.

`AuthHook`'s progression gate now passes (`stepIndex(CONSENT=4) >= stepIndex(CONSENT=4)`), and the user lands on the consent screen as the final pre-completion onboarding step.

**Drive-by fix**: [`test/routes/my-artists-route.spec.ts`](https://github.com/liverty-music/frontend/blob/9bdd79d/test/routes/my-artists-route.spec.ts#L19-L30) mocks `OnboardingStep` with an inline literal map (the test pre-dates this PR). The mock was missing the new `CONSENT: 'consent'` key, which caused the spy assertion to receive `undefined` once the production code became correct. Added the missing key alongside the existing entries.

`make check` passes clean (1113 + 7 = 1120 tests, no failures). I confirmed via `grep` that `setStep(OnboardingStep.COMPLETED)` had only this one call site in `src/` — no other transitions to skip.
